src/Corrade/Utility/Macros.h

```diff
@@ -390,16 +390,38 @@ regarding RAII.
 @brief C++14 constexpr
 @m_since{2020,06}
 
-Expands to @cpp constexpr @ce on C++14 and newer, empty on C++11. Useful for
-selectively marking functions that make use of C++14 relaxed constexpr rules.
+Expands to @cpp constexpr @ce on compilers supporting C++14 relaxed constexpr
+rules. This includes all that fully implement the standard but excludes
+partial implementations such as MSVC 2015 or GCC 4.8.
 @see @ref CORRADE_CXX_STANDARD
 */
-#if CORRADE_CXX_STANDARD >= 201402
+#if (defined __cpp_constexpr && __cpp_constexpr >= 201304) || \
+    (defined CORRADE_TARGET_MSVC && _MSC_VER >= 1910 && CORRADE_CXX_STANDARD >= 201402L)
 #define CORRADE_CONSTEXPR14 constexpr
 #else
 #define CORRADE_CONSTEXPR14
 #endif
 
+/** @hideinitializer
+@brief C++20 is_constant_evaluated
+@m_since{2022,10}
+
+Expands to a predicate determining whether given @cpp constexpr @ce function
+is being evaluated at compile-time or not. Under C++14 rules, constexpr
+functions may be defined as long as at least some of the code paths are able
+to be executed at compile-time. As long as a fallback is present, features
+such as SIMD or inline assembly may be used in functions marked constexpr.
+
+This support is available on all C++20 capable compilers, but also certain
+ones (Clang 9, GCC 9, MSVC 17.1) that expose the feature as a non-portable
+extension. In which case it may be used under C++14 relaxed constexpr rules.
+*/
+#if (defined(__clang__) && __clang_major__ >= 9) || (defined(__GNUG__) && __GNUG__ >= 9) || (defined(_MSC_VER) && _MSC_VER >= 1931)
```

Comment:
```suggestion
#if (defined(CORRADE_TARGET_CLANG) && !defined(CORRADE_TARGET_APPLE_CLANG) && __clang_major__ >= 9) || (defined(CORRADE_TARGET_APPLE_CLANG) && __clang_major__*100 + __clang_minor__ >= 1104) || (defined(CORRADE_TARGET_GCC) && __GNUC__ >= 9) || (defined(CORRADE_TARGET_MSVC) && _MSC_VER >= 1931)
```

This handles also Apple Clang. For clang-cl I checked the CIs and even MSVC 2019 seems to bundle Clang 12, so the check for `_MSC_VER` should suffice.